src/core/lib/promise/sleep.cc

```diff
@@ -54,12 +54,9 @@ Poll<absl::Status> Sleep::operator()() {
 }
 
 Sleep::ActiveClosure::ActiveClosure(Timestamp deadline)
-    : waker_(Activity::current()->MakeOwningWaker()) {
-  auto engine = GetContext<EventEngine>();
-  GPR_ASSERT(engine != nullptr &&
```

Comment:
I know the code looks nicer without the check, but the anonymous crash is harder to debug. IIRC, it is very easy to forget to add a Context in the right place, and challenging to figure out where that context needed to be after a crash due to a missing context. Maybe that's changed, but if you're removing this check and the helpful error output along with it, it might be worth deleting an engine context in a few places and seeing what the errors look like, if they'll be easier or harder to debug (for anyone on the team).

Reply:
This was originally an integration error... but I'm not happy with sacrificing readability here either - and over time we're going to end up with a very large mess of checks and not checks done differently throughout the codebase. So let's do #31205 instead, and keep this as is.